pkg/adsc/adsc_test.go

```diff
@@ -178,50 +178,261 @@ func TestADSC_Save(t *testing.T) {
 			desc: "populated",
 			err:  nil,
 			expectedJSON: map[string]string{
-				"_lds_tcp": `{
-    "listener-1": {
-      "name": "bar",
-      "ListenerSpecifier": null
-    },
-    "listener-2": {
-      "name": "mar",
-      "ListenerSpecifier": null
-    }
-  }`,
-				"_lds_http": `{
-    "http-list-1": {
-      "name": "bar",
-      "ListenerSpecifier": null
-    },
-    "http-list-2": {
-      "name": "mar",
-      "ListenerSpecifier": null
-    }
-  }`,
-				"_rds": `{
-    "route-1": {
-      "name": "mar"
-    }
-  }`,
-				"_eds": `{
-    "load-assignment-1": {
-      "cluster_name": "foo"
-    }
-  }`,
-				"_ecds": `{
-    "eds-cluster-1": {
-      "name": "test",
-      "ClusterDiscoveryType": null,
-      "LbConfig": null
-    }
-  }`,
-				"_cds": `{
-    "cluster-1": {
-      "name": "foo",
-      "ClusterDiscoveryType": null,
-      "LbConfig": null
-    }
-  }`,
+				"_lds_tcp": `{"listener-1":{
+  "name": "bar",
+  "address": null,
+  "stat_prefix": "",
+  "filter_chains": [
+  ],
+  "use_original_dst": null,
```

Comment:
this seems wrong, why are we outputting a bunch of nulls?

Reply:
it's impacted by jsonpb.Marshaler's parameters, I removed them, then no default zero value now.